draft-ietf-tls-esni.md

```diff
@@ -259,109 +274,114 @@ are in use, or if names can be added or removed from the anonymity set during
 
-For clients (in ClientHello), this extension contains the following
-ClientEncryptedCH structure:
+The extension request is carried by the ClientHelloOuter, i.e., the ClientHello
+transmitted to the client-facing server. The payload contains the following
+`ClientEncryptedCH` structure:
 
 ~~~~
-   struct {
-       HpkeCipherSuite suite;
-       opaque record_digest<0..2^16-1>;
+    struct {
+       HpkeCipherSuite cipher_suite;
+       opaque config_digest<0..2^16-1>;
        opaque enc<1..2^16-1>;
        opaque encrypted_ch<1..2^16-1>;
-   } ClientEncryptedCH;
+    } ClientEncryptedCH;
 ~~~~
 
-suite
-: The HpkeCipherSuite cipher suite used to encrypt ClientHelloInner. This MUST
-match a value provided in the corresponding ECHConfig.cipher_suites list.
+cipher_suite
+: The cipher suite used to encrypt ClientHelloInner. This MUST match a value
+provided in the corresponding `ECHConfig.cipher_suites` list.
 
-record_digest
-: A cryptographic hash of the ECHConfig structure from which the ECH key was
+config_digest
+: A cryptographic hash of the `ECHConfig` structure from which the ECH key was
 obtained, i.e., from the first byte of "version" to the end of the structure.
-This hash is computed using the hash function associated with `suite`, i.e., the
-corresponding HPKE KDF algorithm hash.
+This hash is computed using the hash function associated with `cipher_suite`,
+i.e., the corresponding HPKE KDF algorithm hash.
 
 enc
 : The HPKE encapsulated key, used by servers to decrypt the corresponding
-encrypted_ch field.
+`encrypted_ch` field.
 
 encrypted_ch
-: The serialized and encrypted ClientHelloInner structure, AEAD-encrypted using
-HPKE with the selected KEM, KDF, and AEAD algorithm and key generated as
-described below. {:br}
+: The serialized and encrypted ClientHelloInner structure, encrypted using HPKE
+as described in {{send-ech}}.
 
-If the server accepts ECH, it does not send this extension. If it rejects ECH,
-then it sends the following structure in EncryptedExtensions:
+When the client-facing server accepts ECH, it does not send this extension. When
+it rejects, it adds an "encrypted_client_hello" extension to EncryptedExtensions
+with the following structure as the payload:
 
 ~~~
-   struct {
+    struct {
        ECHConfigs retry_configs;
-   } ServerEncryptedCH;
+    } ServerEncryptedCH;
 ~~~
 
 retry_configs
-: An ECHConfigs structure containing one or more ECHConfig structures in
-decreasing order of preference that the client should use on subsequent
-connections to encrypt the ClientHelloInner structure.
+: An ECHConfigs structure containing one or more ECHConfig structures, in
+decreasing order of preference, to be used by the client in subsequent
+connection attempts.
 
-This protocol also defines the "ech_required" alert, which is sent by the client
-when it offered an "encrypted_client_hello" extension which was not accepted by
-the server.
+This document also defines the "ech_required" alert, which clients MUST send
```

Comment:
This is meant to be clarifying, not a spec change. The "ech_required" alert MUST be sent in this case, correct?

Reply:
👍 yep!